src/services/AsyncStorage.ts

```diff
@@ -84,12 +84,38 @@ const setExerciseProgress = async (disciplineId: number, exerciseKey: ExerciseKe
   await AsyncStorage.setItem(PROGRESS_KEY, JSON.stringify(savedProgress))
 }
 
+const calculateScore = (documentsWithResults: DocumentResult[]): number => {
```

Comment:
Not sure if it is the right place here to do something like this. Imo `AsyncStorage` should be just about CRUD operations to the async storage. Not about calculating scores and similar.

Reply:
moved to helpers